pants.toml

```diff
@@ -0,0 +1,456 @@
+# All of the following are seeded with defaults in the config
+#   user: the current user
+#   homedir: the current user's home directory
+#   buildroot: the root of this repo
+#   pants_bootstrapdir: the global pants scratch space primarily used for caches
+#   pants_configdir: the global pants storage for config files
+#   pants_supportdir: pants support files for this repo go here; for example: ivysettings.xml
+#   pants_distdir: user visible artifacts for this repo go here
+#   pants_workdir: the scratch space used to for live builds in this repo
+
+
+# [DEFAULT] should only contain variables to be used later in this config or
+# options not in global scope but available in certain subsystems or tasks.
+[DEFAULT]
+# TODO: Still needed until we migrate jvm tools to subsystems.
+jvm_options = ["-Xmx1g"]
+local_artifact_cache = "%(pants_bootstrapdir)s/artifact_cache"
+
+# [GLOBAL] should only contain valid options in global scope.
+[GLOBAL]
+print_exception_stacktrace = true
+
+# Enable our own custom loose-source plugins as well as contribs.
+pythonpath = [
+  "%(buildroot)s/contrib/avro/src/python",
+  "%(buildroot)s/contrib/awslambda/python/src/python",
+  "%(buildroot)s/contrib/buildrefactor/src/python",
+  "%(buildroot)s/contrib/codeanalysis/src/python",
+  "%(buildroot)s/contrib/cpp/src/python",
+  "%(buildroot)s/contrib/confluence/src/python",
+  "%(buildroot)s/contrib/errorprone/src/python",
+  "%(buildroot)s/contrib/findbugs/src/python",
+  "%(buildroot)s/contrib/go/src/python",
+  "%(buildroot)s/contrib/googlejavaformat/src/python",
+  "%(buildroot)s/contrib/jax_ws/src/python",
+  "%(buildroot)s/contrib/mypy/src/python",
+  "%(buildroot)s/contrib/node/src/python",
+  "%(buildroot)s/contrib/python/src/python",
+  "%(buildroot)s/contrib/scalajs/src/python",
+  "%(buildroot)s/contrib/scrooge/src/python",
+  "%(buildroot)s/contrib/thrifty/src/python",
+  "%(buildroot)s/pants-plugins/src/python",
+]
+backend_packages.add = [
```

Comment:
Can `toml` keys contain any useful special characters? If so, it might actually be desirable to have the ".add"/".remove" syntaxes jump out even more than they do as magical... strawman:
```
backend_packages.++ = [
  ..
]
```

(related: @illicitonion recently pointed out a convention of using `#` keys in json to add comments, which made me think about this.)

Reply:
Oh good idea! Unfortunately, with the current spec, you'd have to wrap it in quotes:

```toml
backend_packages."++" = [
  "..",
]
```

But they're considering loosening the spec to allow any unicode symbol for a key name (particularly helpful for other languages). https://github.com/toml-lang/toml/issues/687